073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,250 @@
+3. The CaReconciler rolls the ZooKeeper and Kafka nodes to trust the new CA certificates
+4. The ZooKeeper and Kafka reconcilers generate new certificates signed by the new CA key and roll the pods to use the new certificates
+5. The EntityOperator, CruiseControl and KafkaExporter reconcilers generate new certificates signed by the new CA key amd roll the pods to use the new certificates
+6. A new reconciliation starts
+7. The CaReconciler updates the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret with the new CA certificate (since the generations are now correct)
+8. The CaReconciler removes the old CA certificates and keys that have been named ca-YYYY-MM-DDTHH-MM-SSZ.crt.
+
+The existing logic assumes two things:
+ - all certificates used by components have been signed by the same root CA certificate
+ - the CaReconciler runs and (if required) has either created or updated the CA certificates and keys before the other reconcilers run, i.e. the CA certificate is already present when e.g. the KafkaReconciler starts
+
+## Motivation
+
+In future it would be beneficial if Strimzi can integrate with commonly deployed certificate management solutions, such as [Cert Manager][cmio], or [Vault][vault].
+These solutions work by having the user request certificates from a particular issuer.
+
+Integrating with these solutions requires a change in how things are managed. Specifically:
+ - Strimzi cannot guarantee that the issuer will issue every certificate signed by the same root CA certificate. 
+   - For example Cert Manager will include the CA certificate to trust when it issues the certificate (if the CA certificate is not a public one)
+   - Using those solutions therefore requires TLS peers to be able to trust multiple root CA certificates.
+ - These solutions cannot be responsible for rolling updates to components like Kafka.
+   - This means that end-entity certificate generation and the rollout of trust in the CA certificates referenced in those end-entity certificates, needs to be decoupled.
+ - Certificates will likely be requested in an asynchronous manner, which means it can't be assumed that e.g. the CA certificate is already present when the component reconcilers run.
+This would be highly valuable for organizations with compliance requirements with regard to certificates.
+
+In addition, currently it is hard to see the current state of trust within the cluster when certificates are being renewed.
+The proposed changes make it easier to reason about the current state and track the roll out of trust.
+
+## Proposal
+
+Strimzi should be updated to decouple the management of the cluster and clients CA certificates, and the consumption of those CA certificates by components (such as ZooKeeper, Kafka and cluster operator).
+
+To do this we will introduce new shared Kubernetes Secrets that are used to keep track of CA certificates that are currently in use and their current state with respect to issued end entity certificates.
```

Comment:
I would give a name to this "new shared Kubernetes Secrets" to make it easier to refer to it across the proposal.